src/core/tokenizer.c

```diff
@@ -188,10 +245,13 @@ tokenize(filter_pipeline_t *fp, const char *text, size_t text_len __unused)
 		}
 		tokenset_add(tset, token);
 	}
-	free(content);
+	ubrk_close(it_token);
+	free(utext);
 	return tset;
 err:
 	tokenset_destroy(tset);
-	free(content);
+	if (utext != NULL) {
```

Comment:
You may pass NULL to `free(3)` (the C standard mandates that it has to check for it).